models/_sector/depin/geodnet/geodnet_revenue.sql

```diff
@@ -0,0 +1,127 @@
+{{ config(
+    schema = 'depin'
+    , alias = 'geodnet_revenue'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['date', 'chain', 'name']
+    )
+}}
+
+WITH
+  hour_series AS (
+    SELECT
+      time AS hour,
+      DATE_TRUNC('day', time) AS day
+    FROM
+      UNNEST (
+        sequence(
+          CAST('2023-04-20 00:00' AS timestamp),
+          CAST(
+            DATE_TRUNC('hour', CURRENT_TIMESTAMP) AS timestamp
+          ),
+          INTERVAL '1' hour
+        )
+      ) AS t (time)
+      {% if is_incremental() %}
+      WHERE
+          time >= (
+            SELECT
+              CAST(COALESCE(MAX(date), '2023-04-27 00:00') AS timestamp) - INTERVAL '7' DAY
```

Comment:
By the way does '2023-04-27 00:00' have special meaning here?
If it is the project start date we introduce a jinja var:
```
{% set project_start_date = '2023-04-20' %}
```

Reply:
ah cool!